tripy/tripy/frontend/ops/equal.py

```diff
@@ -0,0 +1,59 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from tripy import export, constraints
+from tripy.common.exception import raise_error
+
+
+@export.public_api(document_under="operations/functions")
+@constraints.dtype_info(
+    dtype_variables={"T1": ["float32", "float16", "bfloat16", "float8", "int4", "int8", "int32", "int64", "bool"]},
+    dtype_constraints={"a": "T1", "b": "T1"},
+)
+def equal(a: "tripy.Tensor", b: "tripy.Tensor") -> bool:
```

Comment:
If we're following numpy/torch semantics, `allclose` doesn't account for shape while `equal` does. If we want to add `equal`, we should also check for shape below. Otherwise `allclose(..., rtol=0, atol=0)` should be equivalent to this.

Reply:
I see. Yes, I see a few places where we used `bool(tp.all(a == b))` so I thought it would be helpful to add such an op in frontend.
@pranavm-nvidia I will fix equal to check for shape, type, and stride.

Reply:
@pranavm-nvidia I think all.close also compare the shapes.
```
    def test_torch_all_close(self):
        t = torch.arange(12)
        r = t.reshape(3, 4)
        print(t.shape)
        print(r.shape)
>       assert torch.allclose(t, r)
E       RuntimeError: The size of tensor a (12) must match the size of tensor b (4) at non-singleton dimension 1

tests/integration/test_equal.py:98: RuntimeError
---------------------------------------------------------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------------------------------------------------------
torch.Size([12])
torch.Size([3, 4])
```

I will remove `tp.equal` operation for now and use `tp.allclose` where required.

Reply:
Let's add `tp.equal` operation in a separate PR.